### PR TITLE
clone: Use newer Box syntax and Debug trait for printing

### PR DESCRIPTION
**Pull request by badboy**
This runs for me on latest rustc.

### CODE REVIEW

**Comment by steveklabnik** [comment]
Thank you! We should be able to remove the feature gate on line 1 then, too. Mind doing that?

**Comment by badboy** [comment]
aaand done.

File: examples/clone/clone.rs [hunk 1]
```diff
@@ -27,7 +25,7 @@ fn main() {
     println!("copy: {:?}", moved_pair);
 
     // Error! `pair` has lost it resources
-    //println!("original: {}", pair);
+    println!("original: {:?}", pair);
```
Comment (steveklabnik) [inline examples/clone/clone.rs]: when you lost the `//`, it broke the tests!

Comment (badboy) [inline examples/clone/clone.rs]: D'oh!

**Comment by steveklabnik** [comment]
Thanks so much!

**Pull request merged**